process/transaction/shardProcess.go

```diff
@@ -173,6 +173,13 @@ func (txProc *txProcessor) ProcessTransaction(tx *transaction.Transaction) (vmco
 			}
 		}
 
+		if errors.Is(err, process.ErrUserNameDoesNotMatch) && txProc.flagRelayedTx.IsSet() {
+			receiptErr := txProc.executingFailedTransaction(tx, acntSnd, err)
+			if receiptErr != nil {
+				return vmcommon.UserError, receiptErr
+			}
+		}
```

Comment:
Should be retuned (vmcommon.UserError, err) or (vmcommon.UserError, nil) like in L188?

Reply:
It is correct now. ReceiptErr must be returned as it signals ErrFailedTransaction

Reply:
I want to say if it is missed a return between line 180-181: return vmcommon.UserError, nil, or it is correct to use the final return from line 190: return vmcommon.UserError, err ?

Reply:
Yes. It is correct. That tx will not be taken into consideration when building the block.